lib/rspec/core/runner.rb

```diff
@@ -122,7 +124,7 @@ def run_specs(example_groups)
           end
         end && !@world.non_example_failure
 
-        success ? 0 : @configuration.failure_exit_code
+        success ? 0 : @configuration.failure_or_error_exit_code(@world)
```

Comment:
Given the fact we check for `@world.non_example_failure` above, and need to use the shared logic around wether to return a failure or an exit code both here and in other things using the runner, I think that logic belongs in the runner as something like:

```ruby
def exit_code(success = false)
  return @configuration.error_exit_code if @world.non_example_failure
  success ? 0 : @configuration.failure_exit_code
end
```
and we should refactor this to use that rather than adding it in the configuration, likewise this could then be used from the bisector.